beamer/util.py

```diff
@@ -88,9 +87,6 @@ def is_valid_pair(
         target_chain_id: ChainId,
         target_token_address: ChecksumAddress,
     ) -> bool:
-        if os.environ.get("BEAMER_ALLOW_UNLISTED_PAIRS") is not None:
```

Comment:
> nitpick: Why not leaving this statement here? Semantically, `is_valid_pair` should return True if the env variable is set.

That was my initial approach, but I think it's cleaner this way. If we want to have that check inside, we need to return a `Optional[bool]`, which I don't really like. If we add the contract code check inside, we need to pass a `web3` object in, which destroys the purpose of the token match checker.

Reply:
> why do we have to return `Optional[bool]`?

I think it doesn't really show the intention. what does no return value mean?

Reply:
> Okay, but I would still leave the if statement inside regarding the env variable.

What do you mean?

Reply:
> > why do we have to return `Optional[bool]`?
> 
> I think it doesn't really show the intention. what does no return value mean?

I wonder why we even have to do that. Can't we enforce a boolean return value? Or what would be the case when there is no return value?

Reply:
> > Okay, but I would still leave the if statement inside regarding the env variable.
> 
> What do you mean?

What was removed in line 91 inside of is_valid_pair(). I would leave that. Because with removing that condition, if BEAMER_ALLOW_UNLISTED_PAIRS is set it would return False if it is not in the list. This is semantically wrong in my eyes. It still should return True.

Reply:
>I wonder why we even have to do that. Can't we enforce a boolean return value? Or what would be the case when there is no return value?

So my idea here is that we only check the contract code (because it's expensive based on the rpc call it makes) when token match checking is disabled. But just a `bool` return value cannot express this, because true/false only means if the token pair is accepted or not.

Reply:
> > > Okay, but I would still leave the if statement inside regarding the env variable.
> > 
> > 
> > What do you mean?
> 
> What was removed in line 91 inside of is_valid_pair(). I would leave that. Because with removing that condition, if `BEAMER_ALLOW_UNLISTED_PAIRS` is set it would return `False` if it is not in the list. This is semantically wrong in my eyes. It still should return `True`.

I see it more in the sense that the token match checker return whether or not the tokens are in it's list. If that result should be used, is decided by `BEAMER_ALLOW_UNLISTED_PAIRS`. But if you disagree I can  change this to return `Optional[bool``]`

Reply:
> > > > Okay, but I would still leave the if statement inside regarding the env variable.
> > > 
> > > 
> > > What do you mean?
> > 
> > 
> > What was removed in line 91 inside of is_valid_pair(). I would leave that. Because with removing that condition, if `BEAMER_ALLOW_UNLISTED_PAIRS` is set it would return `False` if it is not in the list. This is semantically wrong in my eyes. It still should return `True`.
> 
> I see it more in the sense that the token match checker return whether or not the tokens are in it's list. If that result should be used, is decided by `BEAMER_ALLOW_UNLISTED_PAIRS`. But if you disagree I can change this to return `` Optional[bool``] ``

okay from that point, I agree. I would then rather go with the approach which we had before, returning `bool`

Reply:
so I would revert `333dbdba43101bf89db92e619b9a8a8899393fa1` and we are good to go. Thanks!

Reply:
👍